st2actions/tests/unit/test_scheduler.py

```diff
@@ -231,3 +232,30 @@ def test_processing_when_task_completed(self, mock_execution_queue_delete, mock_
         mock_action_service.update_status.assert_not_called()
         mock_execution_queue_delete.assert_called_once()
         ActionExecutionSchedulingQueue.delete(schedule_q_db)
+
+    @mock.patch('st2actions.scheduler.handler.LOG')
+    def test_failed_next_item(self, mocked_logger):
+        self.reset()
+
+        liveaction_db = self._create_liveaction_db()
+
+        schedule_q_db = self.scheduler._create_execution_queue_item_db_from_liveaction(
+            liveaction_db,
+        )
+
+        schedule_q_db = ActionExecutionSchedulingQueue.add_or_update(schedule_q_db)
+
+        with mock.patch(
+            'st2actions.scheduler.handler.ActionExecutionSchedulingQueue.add_or_update',
+            side_effect=db_exc.StackStormDBObjectWriteConflictError(schedule_q_db)
+        ):
+            schedule_q_db = self.scheduling_queue._get_next_execution()
+            self.assertIsNone(schedule_q_db)
+
+        mocked_logger.info.assert_called_once()
+        call_args = mocked_logger.info.call_args_list[0][0]
```

Comment:
That works for now :+1: 

It's not the most robust test, but writing an end to end test to simulate that would be very hard.